File: VBSample/PortSIP/Session.vb
'!
' * @author Copyright (c) 2008-2020 PortSIP Solutions,Inc. All rights reserved.
' * @version 17
' * @see https://www.portsip.com
' * @brief PortSIP SDK Callback events.
' 
' PortSIP VoIP SDK Callback events description.
' 



Imports System.Collections.Generic
Imports System.Text



Namespace PortSIP
    Class Session
        Private mSessionId As Integer = 0
        Private mHoldState As Boolean = False
        Private mSessionState As Boolean = False
        Private mRecvCallState As Boolean = False
        Private mOriginCallSessionId As Integer = 0
        Private mIsReferCall As Boolean = False
        Private mHasEarlyMedia As Boolean = False
        Private mDialogMessageId As Integer = 0
        Private mOutOfDialogMessageId As Integer = 0

        Public Function hasEarlyMedia() As Boolean
            Return mHasEarlyMedia
        End Function
        Public Sub setEarlyMeida(earlyMedia As Boolean)
            mHasEarlyMedia = earlyMedia
        End Sub

        Public Function isReferCall() As Boolean
            Return mIsReferCall
        End Function

        Public Function getOriginCallSessionId() As Integer
            Return mOriginCallSessionId
        End Function

        Public Sub setReferCall(referCall As Boolean, originCallSessionId As Integer)
            mIsReferCall = referCall
            mOriginCallSessionId = originCallSessionId
        End Sub

        Public Sub setDialogMessageId(id As Integer)
            mDialogMessageId = id
        End Sub

        Public Function getDialogMessageId() As Integer
            Return mDialogMessageId
        End Function

        Public Sub setOutOfDialogMessageId(id As Integer)
            mOutOfDialogMessageId = id
        End Sub

        Public Function getOutOfDialogMessageId() As Integer
            Return mOutOfDialogMessageId
        End Function



        Public Sub reset()
            mSessionId = 0
            mHoldState = False
            mSessionState = False
            mRecvCallState = False
            mOriginCallSessionId = 0
            mIsReferCall = False
            mDialogMessageId = 0
            mOutOfDialogMessageId = 0
        End Sub


        Public Sub setSessionId(sessionId As Integer)
            mSessionId = sessionId
        End Sub


        Public Function getSessionId() As Integer
            Return mSessionId
        End Function

        Public Sub setHoldState(state As Boolean)
            mHoldState = state
        End Sub


        Public Function getHoldState() As Boolean
            Return mHoldState
        End Function

        Public Sub setSessionState(state As Boolean)
            mSessionState = state
        End Sub


        Public Function getSessionState() As Boolean
            Return mSessionState
        End Function



        Public Sub setRecvCallState(state As Boolean)
            mRecvCallState = state
        End Sub


        Public Function getRecvCallState() As Boolean
            Return mRecvCallState
        End Function

    End Class
End Namespace

File: CSharpSample_DLL/PortSIP/Session.cs
using System;
using System.Collections.Generic;
using System.Text;



namespace PortSIP
{
    class Session
    {
        private int mSessionId = 0;
        private bool mHoldState = false;
        private bool mSessionState = false;
        private bool mRecvCallState = false;
        private int mOriginCallSessionId = 0;
        private bool mIsReferCall = false;
        private bool mHasEarlyMedia = false;
        private int mDialogMessageId = 0;
        private int mOutOfDialogMessageId = 0;

        public bool hasEarlyMedia()
        {
            return mHasEarlyMedia;
        }
        public void setEarlyMeida(bool earlyMedia)
        {
            mHasEarlyMedia = earlyMedia;
        }

        public bool isReferCall()
        { 
            return mIsReferCall; 
        }

        public int getOriginCallSessionId() 
        { 
            return mOriginCallSessionId; 
        }

        public void setReferCall(bool referCall, int originCallSessionId)
        {
            mIsReferCall = referCall;
            mOriginCallSessionId = originCallSessionId;
        }

        public void setDialogMessageId(int id)
        {
            mDialogMessageId = id;
        }

        public int getDialogMessageId()
        {
            return mDialogMessageId;
        }

        public void setOutOfDialogMessageId(int id)
        {
            mOutOfDialogMessageId = id;
        }

        public int getOutOfDialogMessageId()
        {
            return mOutOfDialogMessageId;
        }



        public void reset()
        {
            mSessionId = 0;
            mHoldState = false;
            mSessionState = false;
            mRecvCallState = false;
            mOriginCallSessionId = 0;
            mIsReferCall = false;
            mDialogMessageId = 0;
            mOutOfDialogMessageId = 0;
        }


        public void setSessionId(int sessionId)
        {
            mSessionId = sessionId;
        }


        public int getSessionId()
        {
            return mSessionId;
        }

        public void setHoldState(bool state)
        {
            mHoldState = state;
        }


        public bool getHoldState()
        {
            return mHoldState;
        }

        public void setSessionState(bool state)
        {
            mSessionState = state;
        }


        public bool getSessionState()
        {
            return mSessionState;
        }



        public void setRecvCallState(bool state)
        {
            mRecvCallState = state;
        }


        public bool getRecvCallState()
        {
            return mRecvCallState;
        }

    }
}
